src/util/util.c

```diff
@@ -971,6 +971,36 @@ bool is_user_or_group_name(const char *sudo_user_value)
     return true;
 }
 
+bool is_syntactically_correct_domain_name(const char *domain)
+{
+    size_t i;
+    char c;
+
+    if ((domain == NULL) || (domain[0] == 0)) {
+        return false;
+    }
+
+    /* Per man page:
+     * ```
+     * A domain name should only consist of alphanumeric ASCII characters,
+     * dashes, dots and underscores.
+     * ```
+     */
+    for (i = 0; i < strlen(domain); ++i) {
```

Comment:
If there should be some limit about maximum domain name size here?

Reply:
So far I only implemented man page statement. But perhaps this statement itself should be stricter.

Reply:
OK, so maybe this is the story for other PR or contribution proposal.